src/plugins/files/server/blob_storage_service/adapters/es/content_stream/content_stream.ts

```diff
@@ -110,10 +139,22 @@ export class ContentStream extends Duplex {
         chunks.push(chunk);
       }
       const buffer = Buffer.concat(chunks);
-      const source: undefined | FileChunkDocument = buffer.byteLength
-        ? cborx.decode(buffer)?._source
+      const decodedChunkDoc: GetResponse<FileChunkDocument> | undefined = buffer.byteLength
+        ? (cborx.decode(buffer) as GetResponse<FileChunkDocument>)
         : undefined;
 
+      // Because `asStream` was used in retrieving the document, errors will also not be processes
+      // and thus are returned "as is", so we check to see if an ES error occurred while attempting
+      // to retrieve the chunk.
+      if (decodedChunkDoc && ('error' in decodedChunkDoc || !decodedChunkDoc.found)) {
+        const err = new Error(`Failed to retrieve chunk id [${id}] from index [${chunkIndex}]`);
+        this.logger.error(err);
```

Comment:
FYI: I normally prefer `.error()` instead of `.debug` just because it will provide imediate feedback in the logs without requiring a customer to set debugging to `debug`. The code base seems to (for the most part) use `.debug()` so let me know if I should change it.

Reply:
LGTM